distribution.md

```diff
@@ -0,0 +1,180 @@
+
+A buildpack blob MUST be packaged as gzip-compressed tarball.
+Its filename should end in `.tgz`.
+
+A buildpack blob MUST contain a `buildpack.toml` file at its root directory.
+
+A buildpack defined within `buildpack.toml` MUST either be:
+
+1. A buildpack implementation specified by a `stacks` field or
+2. A buildpack order specified by an `order` field referencing other buildpack IDs/versions.
+
+A buildpack reference in `order` MUST either be:
+
+1. A reference to a buildpack inside of the blob using a `path` field ,
+2. A reference to a buildpack outside of the blob using an `id` and `version` field. 
+
+### Buildpackage
+
+A buildpackage MUST exist as either an OCI image on an image registry, an OCI image in a Docker daemon, or a `.cnb` file.
+
+A `.cnb` file MUST be an uncompressed tar archive containing an OCI image. Its file name SHOULD end in `.cnb`.
+
+Each FS layer blob in the buildpackage MUST contain a single buildpack blob and at least one symlink.
+
+A symlink MUST be created for each buildpack version that the blob is assembled to support.
+
+```
+/cnb/blobs/<sha256 checksum of buildpack blob tgz>/
+/cnb/by-id/<buildpack ID1>/<buildpack version> -> /cnb/blobs/<sha256 checksum of blob tgz>/
+/cnb/by-id/<buildpack ID2>/<buildpack version> -> /cnb/blobs/<sha256 checksum of blob tgz>/<subdir>/
+...
+```
+
+A buildpack ID, buildpack version, and at least one stack MUST be provided in the OCI image config as a Label.
+
+Label: `io.buildpacks.buildpackage.metadata`
+```json
+{
+  "id": "<entrypoint buildpack ID>",
+  "version": "<entrypoint buildpack version>",
+  "stacks": [
+    {
+      "id": "<stack ID>",
+      "mixins": ["<mixin name>"]
+    }
+  ]
+}
+```
+
+The buildpack ID and version MUST match a buildpack provided by a layer blob.
+
+For a buildpackage to be valid, each `buildpack.toml` describing a buildpack implementation MUST have all listed stacks.
+
+For each listed stack, all associated buildpacks MUST be a candidate for detection when the entrypoint buildpack ID and version are selected.
+
+Each stack ID MUST only be present once.
+For a given stack, the `mixins` list MUST enumerate mixins such that no included buildpacks are missing a mixin for the stack.
+
+Fewer stack entries as well as additional mixins for a stack entry MAY be specified.
+
+## Data Format
+
+### buildpack.toml (TOML)
+
+```toml
+[buildpack]
+id = "<buildpack ID>"
+name = "<buildpack name>"
+version = "<buildpack version>"
+
+[[order]]
+[[order.group]]
+id = "<buildpack ID>"
+version = "<buildpack version>"
+optional = false
+path = "<path to buildpack>"
+
+[[stacks]]
+id = "<stack ID>"
+mixins = ["<mixin name>"]
+build-images = ["<build image tag>"]
+run-images = ["<run image tag>"]
+
+[metadata]
+# buildpack-specific data
+```
+
+If an order is specified, then `stacks` MUST not be specified.
+
+A buildpack reference inside of a `group` MUST either contain an `id` and `version` or a relative `path`.
```

Comment:
Relative to what?

Reply:
Is there an upper bound on how far up the relative path can go? I'm guessing it can't go outside the buildpack, and while that seems obvious, it would be good to include a MUST NOT clause.

Reply:
Removed this part of the spec.